src/Altinn.App.Api/Controllers/PaymentController.cs

```diff
@@ -0,0 +1,60 @@
+#nullable enable
+using Altinn.App.Api.Infrastructure.Filters;
+using Altinn.App.Core.Features.Payment.Exceptions;
+using Altinn.App.Core.Features.Payment.Models;
+using Altinn.App.Core.Features.Payment.Services;
+using Altinn.App.Core.Internal.Instances;
+using Altinn.App.Core.Internal.Process;
+using Altinn.App.Core.Internal.Process.Elements.AltinnExtensionProperties;
+using Altinn.Platform.Storage.Interface.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Altinn.App.Api.Controllers;
+
+/// <summary>
+/// Controller for handling payment operations.
+/// </summary>
+[AutoValidateAntiforgeryTokenIfAuthCookie]
+[ApiController]
+[ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+[Route("{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/payment")]
+public class PaymentController : Controller
+{
+    private readonly IPaymentService _paymentService;
+    private readonly IInstanceClient _instanceClient;
+    private readonly IProcessReader _processReader;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PaymentController"/> class.
+    /// </summary>
+    public PaymentController(IPaymentService paymentService, IInstanceClient instanceClient, IProcessReader processReader)
+    {
+        _paymentService = paymentService;
+        _instanceClient = instanceClient;
+        _processReader = processReader;
+    }
+
+    /// <summary>
+    /// Get updated payment information for the instance. Will contact the payment provider to check the status of the payment.
+    /// </summary>
+    /// <param name="org">unique identifier of the organisation responsible for the app</param>
+    /// <param name="app">application identifier which is unique within an organisation</param>
+    /// <param name="instanceOwnerPartyId">unique id of the party that this the owner of the instance</param>
+    /// <param name="instanceGuid">unique id to identify the instance</param>
+    /// <returns>An object containing updated payment information</returns>
+    [HttpGet]
+    [ProducesResponseType(typeof(PaymentInformation), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetPaymentInformation(string org, string app, int instanceOwnerPartyId, Guid instanceGuid)
```

Comment:
This endpoint will be called by frontend to get the latest and greatest information about payment for the instance. Ended up using GET, after mild discussions with no strong opinions. Considering the "store payment information" more of a cache than a real state change, but I can be convinced to use a different HTTP verb here. Bring it.